packages/widget/src/pages/SwapPage/SwapPageHeader.tsx

```diff
@@ -1,71 +1,81 @@
-  leftButton,
-  centerButton,
-  rightButton,
-  rightContent,
-}: SwapPageHeaderProps) => {
-  const LeftIcon = iconMap[leftButton?.icon || ICONS.none];
-  const CenterIcon = iconMap[centerButton?.icon || ICONS.none];
-  const RightIcon = iconMap[rightButton?.icon || ICONS.none];
+  const historyPageButton = useMemo(() => {
+    if (transactionHistory?.length === 0) return;
+    const getHistoryPageIcon = () => {
+      if (isFetchingLastTransactionStatus) {
+        return (
+          <div
+            style={{
+              marginLeft: "8px",
+              marginRight: "8px",
+              position: "relative",
+            }}
+          >
+            <SpinnerIcon
+              style={{
+                animation: "spin 1s linear infinite",
+                position: "absolute",
+                height: 14,
+                width: 14,
+              }}
+            />
+          </div>
+        );
+      }
 
-  return (
-    <StyledSwapPageHeaderContainer align="center" justify="space-between">
-      <Row align="center" gap={10}>
-        {leftButton && (
-          <GhostButton gap={5} align="center" onClick={leftButton.onClick}>
-            <LeftIcon />
-            {leftButton.label}
-          </GhostButton>
-        )}
-      </Row>
+      return ICONS.history;
+    };
 
-      <CenterContainer>
-        {centerButton && (
-          <GhostButton gap={5} align="center" onClick={centerButton.onClick}>
-            <CenterIcon />
-            {centerButton.label}
-          </GhostButton>
-        )}
-      </CenterContainer>
+    return {
+      label: "History",
+      icon: getHistoryPageIcon(),
+      onClick: () => {
+        track("swap page: history button - clicked");
+        setCurrentPage(Routes.TransactionHistoryPage);
+      },
+    };
+  }, [isFetchingLastTransactionStatus, setCurrentPage, transactionHistory?.length]);
 
-      <Row align="center" gap={10}>
-        {rightContent}
-        {rightButton && (
-          <GhostButton gap={5} align="center" onClick={rightButton.onClick}>
-            {rightButton.label}
-            <RightIcon />
-          </GhostButton>
-        )}
-      </Row>
-    </StyledSwapPageHeaderContainer>
+  return (
+    <>
+      {isFetchingLastTransactionStatus && <TrackLatestTxHistoryItemStatus />}
+      <PageHeader
+        leftButton={historyPageButton}
+        rightContent={sourceAccount ? <ConnectedWalletContent /> : null}
+      />
+    </>
   );
-};
+});
+
+export const TrackLatestTxHistoryItemStatus = memo(() => {
+  const transactionhistory = useAtomValue(transactionHistoryAtom);
+  const lastTxHistoryItem = transactionhistory.at(-1);
 
-const StyledSwapPageHeaderContainer = styled(Row)`
-  height: 30px;
-  position: relative;
-`;
+  useTxHistory({
+    txHistoryItem: lastTxHistoryItem,
+    index: transactionhistory.length - 1,
+  });
 
-const CenterContainer = styled.div`
-  position: absolute;
-  left: 50%;
-  transform: translateX(-50%);
-`;
+  return null;
+});
```

Comment:
@codingki i'm tracking the latestTxHistoryItemStatus here, conditionally on `isFetchingLastTransactionStatus` which current is 

```ts
  return (
    overallStatus === "pending" ||
    (lastTxHistoryItem?.isSettled !== true && lastTxHistoryItem?.route?.txsRequired === 1)
  );
```
  
  if we remove `overallStatus` it would be what i had before i added overallStatus as a way of make the spinner show up essentially instantly (before the lastTxHistoryItem is updated, because there's currently a delay)
  ```ts
  lastTxHistoryItem?.isSettled !== true && lastTxHistoryItem?.route?.txsRequired === 1
  ```

Reply:
there's a couple unintuitive things here i'm doing to avoid unnecessary re-renders as well as allowing a hook to be conditionally called